libraries/AP_HAL_ChibiOS/CANIface.h

```diff
@@ -97,6 +97,11 @@ class ChibiOS::CANIface : public AP_HAL::CANIface
 
     ChibiOS::bxcan::CanType* can_;
 
+    // state for ISR RX handler. We put this in the class to avoid
+    // having to expand the stack size for all threads
+    AP_HAL::CANFrame isr_rx_frame;
```

Comment:
On hardware with more than one CAN interface, and more than one driver, and possibly 2 or more ISR handlers,  will this mean they all point to the same class-based buffer, and so if both ISRs overlap , they might trash each others isr_rx_frame..?

Reply:
it is a class member, not static, so it's OK from that point of view, plus you can't recurse to the same level of interrupt from within an ISR